nupic/research/frameworks/continuous_learning/dendrite_layers.py

```diff
@@ -124,51 +127,72 @@ class DendriteLayer(nn.Module):
 
 
     """
+
     def __init__(self,
                  in_dim,
                  out_dim,
                  dendrites_per_neuron,
                  weight_sparsity=0.2,
-                 k_inference_factor=1.,
-                 boost_strength=2.,
-                 boost_strength_factor=0.9,
-                 duty_cycle_period=1000,
+                 act_fun_type=None,
                  ):
         super(DendriteLayer, self).__init__()
 
         self.dendrites_per_neuron = dendrites_per_neuron
         self.n_dendrites = out_dim * self.dendrites_per_neuron
         self.out_dim = out_dim
 
+        self.act_fun_type = act_fun_type
+
         self.input = DendriteInput(
             in_dim=in_dim,
             n_dendrites=self.n_dendrites,
             weight_sparsity=weight_sparsity,
         )
         self.output = DendriteOutput(out_dim, self.dendrites_per_neuron)
 
-        self.act_fun = KWinners2d(
-            channels=self.out_dim,
-            percent_on=0.1,  # will be overwritten
-            k_inference_factor=k_inference_factor,  # will be overwritten
-            boost_strength=boost_strength,
-            boost_strength_factor=boost_strength_factor,
-            duty_cycle_period=duty_cycle_period,
-            local=True,
-        )
-        self.act_fun.k = 1
-        self.act_fun.k_inference = 1
+        if self.act_fun_type == "kwinner":
+            self.act_fun = DendriteKWinners2d(
+                channels=self.out_dim,
+                k=1,
+                local=True,
+            )
+            self.act_fun.k = 1
```

Comment:
Why set k to 1 right after are initializing it with k=1?

Reply:
serial editing brain.. will remove